free-2.md

```diff
@@ -58,12 +95,22 @@ making language design decisions), as opposed to "systems" aspects of the
 project (e.g., implementing a complicated semantics that doesn't require a lot
 of language design)?_
 
+I think it would be a 60-40 split in favor of language design.  With a very 
+rigid syntax and structure, it would take a lot of work to get that part down
+exactly.  I think the NLP and parsing would be very non-trivial, but the language
+design would trump that.
+
 
 ### Scope
 _How big an idea is this? How ambitious is this project?_
 
+Medium sized.  I think it would be challenging to do, but manageable for someone
+who knew and enjoyed NLP.
+
 
 ### Benefits and drawbacks
 _Why might this be a good idea for a project? Why might this not be a good idea 
 project?_
 
+It could be a good idea because it should be pretty design-heavy.  It might be a 
+bad idea because it feels like a relatively small project.
```

Comment:
I think there's a lot of interesting potential for a language here. The NLP part is tricky, and focusing on it might lead to a project that's more NLP than DSLs. That might be the right way to go in general; but since this is a class about DSLs... :). In practice, it might mean that--for this class--most of your work focuses on the intermediate representation and semantics, leaving the NLP / front-end for later. If you're interested in choosing this project, develop the ideas a bit more and focus on what "an exact recipe" means and on how a computer might identify (from information in the intermediate representation) whether a recipe is too ambiguous.